cos-fleetshard-sync/src/main/java/org/bf2/cos/fleetshard/sync/resources/cleanup/AddonConfigMapWatcher.java

```diff
@@ -0,0 +1,91 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+
+public class AddonConfigMapWatcher extends AbstractWatcher {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AddonConfigMapWatcher.class);
+
+    private static final String ADDON_ID = "connectors-operator"; // FIXME config?
+    private static final String ADDON_LABEL = "api.openshift.com/addon-connectors-operator";
+    private static final String ADDON_LABEL_DELETE = ADDON_LABEL + "-delete";
+    private static final String OPERATORS_API = "operators.coreos.com/v1alpha1";
+    private static final String OPERATOR_KIND = "ClusterServiceVersion";
+
+    @Inject
+    KubernetesClient kubernetesClient;
+    @Inject
+    FleetShardSyncConfig config;
+    @Inject
+    FleetShardSync fleetShardSync;
+
+    @Override
+    protected Watch doWatch() {
+        LOGGER.info("Creating Watcher for Addon ConfigMaps on namespace {} with Addon ID {}", config.namespace(), ADDON_ID);
+        return kubernetesClient
+                .configMaps()
+                .inNamespace(config.namespace())
+                .withName(ADDON_ID)
+                .watch(this);
+    }
+
+    @Override
+    protected void onEventReceived(Action action, Object resource) {
+        LOGGER.info("Event received for Addon ConfigMap with action {}", action.name());
+        switch (action) {
+            case ADDED:
+            case MODIFIED:
+                cleanupCluster((ConfigMap) resource);
+                break;
+            case DELETED:
+            case ERROR:
+            case BOOKMARK:
+                LOGGER.debug("Event {} not covered by this watcher, nothing will be done.", action.name());
+                break;
+        }
+    }
+
+    private void cleanupCluster(ConfigMap configMap) {
+        String deleteLabel = configMap.getMetadata().getLabels().get(ADDON_LABEL_DELETE);
+        if ("true".equals(deleteLabel)) {
+            String clusterId = config.cluster().id();
+            LOGGER.info("ConfigMap for deletion of the Addon was found, starting cleanup of cluster: {} - {}", clusterId, configMap);
+
+            fleetShardSync.stop();
+
+            LOGGER.info("Deleting all ManagedConnector resources that belong to cluster {}", clusterId);
+            kubernetesClient.resources(ManagedConnector.class)
+                    .inAnyNamespace()
+                    .withLabel(Resources.LABEL_CLUSTER_ID, clusterId)
+                    .delete();
+
+            LOGGER.info("Deleting all Namespaces that belong to cluster {}", clusterId);
+            kubernetesClient.namespaces()
+                    .withLabel(Resources.LABEL_CLUSTER_ID, clusterId)
+                    .delete();
+
+            LOGGER.info("Deleting {} in namespace {} to signal that addon deletion should continue", OPERATOR_KIND, config.namespace());
+            Boolean deleteCsvOp = kubernetesClient.genericKubernetesResources(OPERATORS_API, OPERATOR_KIND)
```

Comment:
is this really needed ? 
is so there should be a related cluster role

Reply:
Yes, this is what signals that we have finished deleting other namespaces and that the addon should continue with the normal deletion process.